include/clap/ext/draft/location.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include "../../plugin.h"
+#include "../../color.h"
+
+// This extension allows a host to tell the plugin more about its position 
+// within a project or session. 
+
+static CLAP_CONSTEXPR const char CLAP_EXT_LOCATION[] = "clap.location/1";
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum {
+   CLAP_PLUGIN_LOCATION_PROJECT,
+   CLAP_PLUGIN_LOCATION_TRACK,
+   CLAP_PLUGIN_LOCATION_TRACK_GROUP,
+   CLAP_PLUGIN_LOCATION_DEVICE,
+   CLAP_PLUGIN_LOCATION_DEVICE_PARALLEL_GROUP,
+   CLAP_PLUGIN_LOCATION_DEVICE_SERIAL_GROUP,
+};
+
+typedef struct clap_plugin_location_element {
+   // Internal ID of the element. This is not intended for display to the user, 
+   // but rather to give the host a potential quick way for lookups.
+   const char *id;
+   // User friendly name of the element.
+   const char *name;
+   // Kind of the element, must be one of the CLAP_PLUGIN_LOCATION_* values.
+   int kind;
+   // Index within the parent element. 
+   uint32_t index_in_group;
+   // Index of the element in a global context. When sorting elements by this index, 
+   // the order should mirror the order as it is displayed in the host. 
+   uint32_t global_index;
```

Comment:
I'm not sure how to provide this value.

You can already sort using the index in group, why would you need a global one?
Keeping the global one in sync, requires to constantly invalidate it after each insertion/deletion in the document.
And I don't see it necessary.

Reply:
I meant the global_index.

Reply:
That's true, but the flipside is that if a plugin wants to, for example, display instances in a flat list in the same order as in the host not having a global index would significantly complicate that sorting process. I understand that for a host it's more work, but it would make host->plugin consistency a lot easier, I think.

Reply:
The complexity for the plugin is super easy. You just implement a compare function that compares the path elements one by one.

While for the host, it is really a difficult job. You're basically asking for the host to invalidate and crawl the entire project on every device insertion/deletion. This hooks into the undo etc...

Reply:
Sorry it took a second, thanks for the explanation, I've changed it

Reply:
I'll re-review the whole ext soon, but I feel that we're good for a merge! :+1: